internal/host/validator.go

```diff
@@ -795,3 +795,21 @@ func (v *validator) printSufficientPacketLossRequirementForRole(c *validationCon
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) isDomainNameResolved(c *validationContext) ValidationStatus {
+	if c.host.DomainNameResolution == c.cluster.APIVip {
```

Comment:
Isn't the `DomainNameResolution` a serialized JSON? I believe you need to deserialize it here, it has a format that looks something like:

```json
{
  "resolutions": [
    {
      "domain_name": "example.com",
      "ipv4_addresses": [
        "93.184.216.34"
      ],
      "ipv6_addresses": [
        "2606:2800:220:1:248:1893:25c8:1946"
      ]
    },
    {
      "domain_name": "example.net",
      "ipv4_addresses": [
        "93.184.216.34"
      ],
      "ipv6_addresses": [
        "2606:2800:220:1:248:1893:25c8:1946"
      ]
    },
    {
      "domain_name": "example.org",
      "ipv4_addresses": [
        "93.184.216.34"
      ],
      "ipv6_addresses": [
        "2606:2800:220:1:248:1893:25c8:1946"
      ]
    }
  ]
}
```

Then you'll have to find the domain name we're interested in validating, and compare the address to what we're expecting - either with IPv4 or IPv6

Reply:
Changed to the right format